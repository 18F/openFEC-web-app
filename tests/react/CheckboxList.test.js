const rendered = require('react-test-renderer');
const React = require('react');
const ReactDOM = require('react-dom');
const CheckboxList = require('../../static/js/legal/filters/CheckboxList');

it('renders checkbox list', () => {
  const documentTypes = [{text: 'Final Opinion', value: 'F'}, {text: 'Request', value: 'R'}];
  const component = rendered.create(
    <CheckboxList key="ao_category" name="ao_category" label="Document Type" value={['F']}
      handleChange={() => {}} options={documentTypes}/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
