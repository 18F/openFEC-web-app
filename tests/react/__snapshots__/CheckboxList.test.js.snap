// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders checkbox list 1`] = `
<div
  className="filter"
>
  <legend
    className="label"
  >
    Document Type
  </legend>
  <div>
    <input
      checked={true}
      id="ao_category_F"
      name="ao_category"
      onChange={[Function]}
      type="checkbox"
      value="F"
    />
    <label
      htmlFor="ao_category_F"
    >
      Final Opinion
    </label>
  </div>
  <div>
    <input
      checked={false}
      id="ao_category_R"
      name="ao_category"
      onChange={[Function]}
      type="checkbox"
      value="R"
    />
    <label
      htmlFor="ao_category_R"
    >
      Request
    </label>
  </div>
</div>
`;
